### PR TITLE
LPS-157830

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://issues.liferay.com/browse/LPS-157830

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java [hunk 1]
```diff
@@ -332,6 +332,14 @@ public Page<ObjectEntry> getObjectEntries(
 				accountEntries, AccountEntry::getAccountEntryId);
 		}
 
+		int start = QueryUtil.ALL_POS;
```
Comment (rodrigopaulino) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:335]: Should this be ordered alphabetically?

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 2]
```diff
@@ -207,13 +209,37 @@ private Predicate _contains(Object fieldName, Object fieldValue) {
...
+				StringUtil.endsWith(
+					objectField.getName(),
+					StringPool.UNDERLINE + objectFieldName)) {
+
+				return objectField;
+			}
+		}
+
+		throw new NoSuchObjectFieldException(
+			"No ObjectField exists with the name " + objectFieldName);
```
Comment (rodrigopaulino) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:247]: Can we turn "ObjectField" into "object field" instead?

**Review by rodrigopaulino — CHANGES_REQUESTED** [review]
Thanks for the quick fix, @carolmariaabb! 🙏🏻 
I have just a couple of suggestions.

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java (hunk 1 above)
Comment (carolmariaabb) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java:335]: In the portal, I found similar cases where they are sorted alphabetically but I also found cases where they are sorted by logic (as I did), like in [here](https://github.com/liferay/liferay-portal/blob/dd3405ce5fe38c8989049dbd4274971e0ffdb0f1/modules/apps/archived/sync-service/src/main/java/com/liferay/sync/service/impl/SyncDLObjectServiceImpl.java#L707-L709). 

I also found this [thread](https://liferay.slack.com/archives/C1SJ64S9X/p1612519027001100) but I still don't now what's the right way.

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java (hunk 2 above)
Comment (carolmariaabb) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:247]: Sure!

Comment (carolmariaabb) [reply modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java]: Done!

**Review by rodrigopaulino — APPROVED** [review]
Since you've found cases like this. We can move on with that and see.
LGTM! Thanks for working on this, @carolmariaabb

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
ci:forward

**Pull request closed without merging**